packages/vite/src/node/plugins/css.ts

```diff
@@ -1800,10 +1938,14 @@ async function minifyCSS(
     const { code, warnings } = await transform(css, {
       loader: 'css',
       target: config.build.cssTarget || undefined,
+      sourcefile: filename || cssBundleName,
       ...resolveMinifyCssEsbuildOptions(config.esbuild || {}),
     })
     if (warnings.length) {
-      const msgs = await formatMessages(warnings, { kind: 'warning' })
+      const msgs = await formatMessages(
+        formatEsbuildWarning(warnings, concatCssEndLines) as Message[],
```

Comment:
```suggestion
        formatEsbuildWarning(warnings, concatCssEndLines),
```
`formatEsbuildWarning` always returns `Message[]` now.